src/encoded/auth0.py

```diff
@@ -96,6 +92,47 @@ def forget(self, request):
         return []
 
 
+def _signup(request):
+    """
+    Create new user.
+
+    :param request: Pyramid request object
+    """
+    domain = 'encode.auth0.com'
+    access_token = request.json['accessToken']
+    url = "https://{domain}/userinfo?access_token={access_token}".format(domain=domain, access_token=access_token)
+    user_data_request = requests.get(url)
+
+    if user_data_request.status_code != 200:
+        raise HTTPBadRequest(explanation='Could not get user data')
+
+    user_data = user_data_request.json()
+
+    if user_data['email_verified'] is not True:
+        raise HTTPBadRequest(explanation='Unverified e-mail')
+
+    if user_data.get('name', ''):
```

Comment:
Seems like this code might break easily? What if someone has two last names? Also are we really sure we want to add 'N/A' to the database?

Reply:
This is tricky.

Reply:
In any case this logic should be split into its own function and unit tested.